src/parcsr_ls/HYPRE_parcsr_ls.h

```diff
@@ -223,21 +223,38 @@ HYPRE_Int HYPRE_BoomerAMGSetMinCoarseSize(HYPRE_Solver solver,
 HYPRE_Int HYPRE_BoomerAMGSetMaxLevels(HYPRE_Solver solver,
                                       HYPRE_Int    max_levels);
 
+/**
+ * (Optional) Sets cut factor for choosing isolated points
+ * during coarsening according to the rows' density. The default is 0.
+ * If nnzrow > coarsen_cut_factor*avg_nnzrow, where avg_nnzrow is the
+ * average number of nonzeros of the global matrix, holds for a given row,
+ * it is set as fine, and interpolation weights are not computed.
+ **/
+HYPRE_Int HYPRE_BoomerAMGSetCoarsenCutFactor(HYPRE_Solver solver,
+                                             HYPRE_Int    coarsen_cut_factor);
+
 /**
  * (Optional) Sets AMG strength threshold. The default is 0.25.
- * For 2d Laplace operators, 0.25 is a good value, for 3d Laplace
+ * For 2D Laplace operators, 0.25 is a good value, for 3D Laplace
  * operators, 0.5 or 0.6 is a better value. For elasticity problems,
- * a large strength threshold, such as 0.9, is often better. The
- * strong threshold for R is strong connections used in building an
- * approximate ideal restriction, and the filter threshold for R a
- * threshold to eliminate small entries from R after building it.
+ * a large strength threshold, such as 0.9, is often better.
  **/
 HYPRE_Int HYPRE_BoomerAMGSetStrongThreshold(HYPRE_Solver solver,
                                             HYPRE_Real   strong_threshold);
 
+/**
```

Comment:
@ulrikeyang should review which of these functions we want to promote to the reference manual.  We don't always want to advertise interface routines to the general user.